Zend/zend_string.h

```diff
@@ -341,30 +341,25 @@ static zend_always_inline void zend_string_efree(zend_string *s)
 	efree(s);
 }
 
-static zend_always_inline void zend_string_release(zend_string *s)
-{
-	if (!ZSTR_IS_INTERNED(s)) {
-		if (GC_DELREF(s) == 0) {
-			pefree(s, GC_FLAGS(s) & IS_STR_PERSISTENT);
-		}
-	}
-}
-
 static zend_always_inline void zend_string_release_ex(zend_string *s, bool persistent)
 {
 	if (!ZSTR_IS_INTERNED(s)) {
 		if (GC_DELREF(s) == 0) {
 			if (persistent) {
 				ZEND_ASSERT(GC_FLAGS(s) & IS_STR_PERSISTENT);
-				free(s);
 			} else {
 				ZEND_ASSERT(!(GC_FLAGS(s) & IS_STR_PERSISTENT));
-				efree(s);
 			}
```

Comment:
Surely the remainder of the if-else block should now be converted to a ZEND_ASSERT?

Reply:
If we did that, I'd prefer to just keep the if/else. The alternative is not very readable. I.e.

```c
ZEND_ASSERT(persistent ? (GC_FLAGS(s) & IS_STR_PERSISTENT) : !(GC_FLAGS(s) & IS_STR_PERSISTENT));
// or
ZEND_ASSERT(persistent == !!(GC_FLAGS(s) & IS_STR_PERSISTENT));
```

Reply:
We could also just wrap the if-else with an `#ifdef ZEND_DEBUG` for readability

Reply:
returned this function to how it was previously.